File: jobradar/js/data/resume.js
/**
 * data/resume.js — 简历列表数据层
 */
import { CONFIG } from '../config.js';

const RESUMES = [
  { id: 1, name: '互联网通用版', updatedAt: '2026-06-10', size: '1.2 MB', color: 'blue',  tags: ['后端', '算法'], active: true },
  { id: 2, name: '金融行业版',   updatedAt: '2026-05-28', size: '0.9 MB', color: 'green', tags: ['量化', '风控'], active: false },
];

const resumeMock = {
  async getAll() { return RESUMES.map((r) => ({ ...r })); },
};

const resumeHttp = {
  async getAll() {
    const res = await fetch(CONFIG.API_BASE + '/resumes');
    if (!res.ok) throw new Error('请求失败 ' + res.status);
    return res.json();
  },
};

export const ResumeStore = CONFIG.USE_MOCK ? resumeMock : resumeHttp;

File: backend/src/main/java/com/jobradar/entity/Resume.java
package com.jobradar.entity;

import jakarta.persistence.*;
import java.util.ArrayList;
import java.util.List;

/** 简历。字段与前端 JSON 一致。 */
@Entity
@Table(name = "resume")
public class Resume {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @Column(nullable = false)
    private Long userId;       // 所属用户（数据隔离）

    private String name;
    private String updatedAt;
    private String size;
    private String color;      // blue / green，决定图标与标签配色
    private boolean active;

    @ElementCollection(fetch = FetchType.EAGER)
    @CollectionTable(name = "resume_tag", joinColumns = @JoinColumn(name = "resume_id"))
    @Column(name = "tag")
    private List<String> tags = new ArrayList<>();

    public Resume() {}

    public Resume(String name, String updatedAt, String size, String color,
                  boolean active, List<String> tags) {
        this.name = name;
        this.updatedAt = updatedAt;
        this.size = size;
        this.color = color;
        this.active = active;
        this.tags = tags;
    }

    // ── getters / setters ──
    public Long getId() { return id; }
    public void setId(Long id) { this.id = id; }
    public Long getUserId() { return userId; }
    public void setUserId(Long userId) { this.userId = userId; }
    public String getName() { return name; }
    public void setName(String name) { this.name = name; }
    public String getUpdatedAt() { return updatedAt; }
    public void setUpdatedAt(String updatedAt) { this.updatedAt = updatedAt; }
    public String getSize() { return size; }
    public void setSize(String size) { this.size = size; }
    public String getColor() { return color; }
    public void setColor(String color) { this.color = color; }
    public boolean isActive() { return active; }
    public void setActive(boolean active) { this.active = active; }
    public List<String> getTags() { return tags; }
    public void setTags(List<String> tags) { this.tags = tags; }
}
